### PR TITLE
Updates VMware documentation links and fixes typos

### PR DESCRIPTION
**Pull request by venilnoronha**
This PR adds a redirection for the VMware vSphere getting started guide to VMware's external documentation [website](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/) and updates related hyperlinks. Also, typos at several locations have been fixed.

This PR produces the following changes:
- Adds `301` redirection for `/docs/getting-started-guides/vsphere/` to `https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/`
- Updates hyperlinks from `/docs/getting-started-guides/vsphere/` to `https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/`
- Fixes typos i.e. `VMWare` and `Vmware` -> `VMware`

### CODE REVIEW

File: docs/setup/pick-right-solution.md [hunk 1]
```diff
@@ -115,9 +115,9 @@ These solutions are combinations of cloud providers and operating systems not co
 
 * [Vagrant](/docs/getting-started-guides/coreos/) (uses CoreOS and flannel)
 * [CloudStack](/docs/getting-started-guides/cloudstack/) (uses Ansible, CoreOS and flannel)
-* [Vmware vSphere](/docs/getting-started-guides/vsphere/)  (uses Debian)
-* [Vmware vSphere, OpenStack, or Bare Metal](/docs/getting-started-guides/ubuntu/) (uses Juju, Ubuntu and flannel)
-* [Vmware](/docs/getting-started-guides/coreos/)  (uses CoreOS and flannel)
+* [VMware vSphere](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/) (uses Debian)
```
Comment (abrarshivani) [inline docs/setup/pick-right-solution.md]: I think it is better to remove ```uses Debian```

File: docs/setup/pick-right-solution.md [hunk 2]
```diff
@@ -164,14 +164,14 @@ GCE                  | CoreOS       | CoreOS | flannel     | [docs](/docs/gettin
 Vagrant              | CoreOS       | CoreOS | flannel     | [docs](/docs/getting-started-guides/coreos/)                                 |  Community ([@pires](https://github.com/pires), [@AntonioMeireles](https://github.com/AntonioMeireles))
 Bare-metal (Offline) | CoreOS       | CoreOS | flannel     | [docs](/docs/getting-started-guides/coreos/bare_metal_offline/)              |  Community ([@jeffbean](https://github.com/jeffbean))
 CloudStack           | Ansible      | CoreOS | flannel     | [docs](/docs/getting-started-guides/cloudstack/)                             |  Community ([@sebgoa](https://github.com/sebgoa))
-Vmware vSphere       | Saltstack    | Debian | OVS         | [docs](/docs/getting-started-guides/vsphere/)                                |  Community ([@imkin](https://github.com/imkin))
+VMware vSphere       | Saltstack    | Debian | OVS         | [docs](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/)  |  Community ([@imkin](https://github.com/imkin))
```
Comment (abrarshivani) [inline docs/setup/pick-right-solution.md]: Let's update this to latest for other fields too

File: docs/setup/pick-right-solution.md (hunk 1 above)
Comment (venilnoronha) [reply docs/setup/pick-right-solution.md]: Addressed in 67b461f.

File: docs/setup/pick-right-solution.md (hunk 2 above)
Comment (venilnoronha) [reply docs/setup/pick-right-solution.md]: Addressed in 67b461f.

**Review by abrarshivani — APPROVED** [review]
LGTM

**Pull request closed without merging**